### PR TITLE
Benchmark all

### PR DESCRIPTION
**Pull request by NicklasHugoy**

### CODE REVIEW

File: src/cubool/cubool_analysis.cpp [hunk 1]
```diff
@@ -14,10 +14,10 @@ GpuBoolMatrix run_analysis(const BoolMatrix& initial, const std::vector<BoolMatr
     std::vector<GpuBoolMatrix> slicers;
     std::vector<GpuBoolMatrix> transfers;
 
+    Stopwatch matrix_alloc_watch;
```
Comment (thorulf4) [inline src/cubool/cubool_analysis.cpp:17]: weeeeell i meaaaan this watch doesnt just watch allocation but also copying :p

File: src/cubool/cubool_analysis.cpp [hunk 2]
```diff
@@ -36,13 +36,14 @@ GpuBoolMatrix run_analysis(const BoolMatrix& initial, const std::vector<BoolMatr
     GpuBoolMatrix next_state(state.rowCount, state.columnCount);
     GpuBoolMatrix slice(state.rowCount, 1);
     GpuBoolMatrix expanded(state.rowCount, state.columnCount);
+    matrix_alloc_watch.printTimeMicroseconds("Matrix allocation ");
```
Comment (thorulf4) [inline src/cubool/cubool_analysis.cpp:39]: printTimeMicroseconds coud have been => print_time<Microsecond>()

File: src/matrix_analysis.cu [hunk 3]
```diff
@@ -53,21 +53,27 @@ void gpu_analysis(std::vector<std::shared_ptr<Node<std::set<std::string>>>>& nod
...
 
 GpuMatrix<float> analyse(std::vector<Matrix<float>>& transfer_matrices, Matrix<float>& successor_matrix, Matrix<float>& initial_state){
-    
-    timeFunc("Cublas creation: ", 
-        create_cublas);
 
     Stopwatch stopwatch;
     
     std::vector<GpuMatrix<float>> transfers;
+    Stopwatch succWatch;
```
Comment (thorulf4) [inline src/matrix_analysis.cu:60]: succ_watch

File: src/main.cpp [hunk 4]
```diff
@@ -77,14 +82,44 @@ int main(int argc, char *argv[]){
...
             if(cubool_flag){
                 std::cout << "Running analysis using cuBool on GPU" << std::endl;
                 auto program = parse_to_cfg_transformer<std::set<std::string>>(prog);
+                timeFunc("Cubool creation: ", 
+                    create_cubool);
                 cubool_analyse(program.nodes);
             }else{
                 std::cout << "Running analysis using cuBLAS on GPU" << std::endl;
                 auto program = parse_to_cfg_transformer<std::set<std::string>>(prog);
+                timeFunc("Cublas creation: ",
```
Comment (thorulf4) [inline src/main.cpp:121]: time_func

Comment (thorulf4) [inline src/main.cpp:121]: could have been time_func<Milisecond> :⭕

**Review by thorulf4 — APPROVED** [review]

**Review by ZcortezZ — APPROVED** [review]

**Pull request merged**